src/java/com/zimbra/oauth/handlers/impl/YahooOAuth2Handler.java

```diff
@@ -141,6 +146,7 @@ public YahooOAuth2Handler(Configuration config) {
         authenticateUri = YahooConstants.AUTHENTICATE_URI;
         authorizeUriTemplate = YahooConstants.AUTHORIZE_URI_TEMPLATE;
         relayKey = YahooConstants.RELAY_KEY;
+        typeKey = YahooConstants.TYPE_KEY;
```

Comment:
Set this in the `OAuth2Handler` constructor or it will be `null` for the other clients.